source/extensions/filters/network/dubbo_proxy/router/router_impl.cc

```diff
@@ -117,6 +117,43 @@ FilterStatus Router::onMessageDecoded(MessageMetadataSharedPtr metadata, Context
   return upstream_request_->start();
 }
 
+void Router::setEncoderFilterCallbacks(DubboFilters::EncoderFilterCallbacks& callbacks) {
+  encoder_callbacks_ = &callbacks;
+}
+
+FilterStatus Router::onMessageEncoded(MessageMetadataSharedPtr metadata, ContextSharedPtr) {
+  if (metadata->hasResponseStatus() && upstream_request_ != nullptr) {
```

Comment:
nit: probably the code would be less nested and more readable if we returned early here:
```c++
FilterStatus Router::onMessageEncoded(MessageMetadataSharedPtr metadata, ContextSharedPtr) {
  if (!metadata->hasResponseStatus() || upstream_request_ == nullptr) {
    return FilterStatus::Continue;
  }

  ENVOY_STREAM_LOG(trace, "dubbo router: response status: {}", *encoder_callbacks_,
                   metadata->responseStatus());
  ...
  return FilterStatus::Continue;
}
```